examples/blocks/js/app.js

```diff
@@ -0,0 +1,130 @@
+(function () {
+	/*global blocks */
+
+	'use strict';
+
+	var ENTER_KEY = 13;
+	var ESCAPE_KEY = 27;
+
+	var App = blocks.Application();
+
+	var Todo = App.Model({
+		title: App.Property(),
+
+		completed: App.Property(),
+
+		editing: blocks.observable(),
+
+		init: function () {
+			var collection = this.collection();
+			if (collection) {
+				this.title.on('change', collection.save);
+				this.completed.on('change', collection.save);
+			}
+			this.title.on('change', function (newValue) {
+				this.title((newValue || '').trim());
+			});
+		},
+
+		toggleComplete: function () {
+			this.completed(!this.completed());
+		},
+
+		edit: function () {
+			this.lastValue = this.title();
+			this.editing(true);
+		},
+
+		closeEdit: function () {
+			if (this.title()) {
+				this.editing(false);
+			} else {
+				this.destroy();
+			}
+		},
+
+		handleAction: function (e) {
+			if (e.which === ENTER_KEY) {
+				this.closeEdit();
+			} else if (e.which === ESCAPE_KEY) {
+				this.title(this.lastValue);
+				this.editing(false);
+			}
+		}
+	});
+
+	var Todos = App.Collection(Todo, {
+		remaining: blocks.observable(),
+
+		init: function () {
+			this
+				.reset(JSON.parse(localStorage.getItem('todos-jsblocks')) || [])
+				.on('add remove', this.save)
+				.updateRemaining();
+		},
+
+		toggleAll: function () {
+			var complete = this.remaining() === 0 ? false : true;
+			this.each(function (todo) {
+				todo.completed(complete);
+			});
+		},
+
+		clearCompleted: function () {
+			this.removeAll(function (todo) {
```

Comment:
Might be just my feeling but `removeAll` is a confusing name if you only want to remove specific items.